### PR TITLE
Fskip sound fix

### PR DESCRIPTION
**Pull request by ARoxdale**
This is a proposed fix for the current PushSampesRequestBufs() overflow issue in Meka.

Essentially the problem appears to have been due to the output sound sample count needing to be based on the output (fskipper) FPS rate. With this, the sound runs smoothly even up to 400Hz, and down to 20Hz. 10Hz still seems to have some issues.

I think this will allow 50Hz games to run properly, even at 60Hz.
I've only tested this on SMS games by the way so you might want to test on other machines.
Also, I've only tested on linux. 

I might try some other code cleanups of the sound code later.

### CODE REVIEW

**Comment by maxim-zhao** [comment]
I guess the force-push has confused this PR, can you rebase?

File: meka/srcs/sound/sound.cpp [hunk 1]
```diff
@@ -378,12 +392,23 @@ void SoundStream_RenderUpToCurrentTime(t_sound_stream* stream)
...
+    //The OUTPUT (Wall) sound clock (in Z80cycles/sec) is different from the
+    //emulated sound clock (in Z80cycles/sec) as we may be running at a non-standard FPS (e.g. 10Hz or 200Hz).
+    //We are generating samples for the OUPUT  sound buffers, so we need to base our clock rate and rendered sample
+    //count on the output FPS rate (otherwise we will over/underflow the actual output sound)
+    
+    //We can base the output FPS rate on either fskipper.Throttled_Speed (requested) or fskipper.FPS (measured)
+    //Throttled speed appears to avoid most overflows, but the actually measured FPS rate does better at 10Hz (and might work for unthrottled fps)
+    const double output_FPS = (double) fskipper.Throttled_Speed;
+    const double output_cycles_per_sec = GetZ80CyclesPerFrame(&g_machine) * output_FPS;
+    const double elasped_output_seconds = (double)((double)elapsed_cycles / (double)output_cycles_per_sec);
```
Comment (maxim-zhao) [inline meka/srcs/sound/sound.cpp]: A bit keen with the casting here?

**Comment by ARoxdale** [comment]
I think the PR should be rebased now.

File: meka/srcs/sound/sound.cpp (hunk 1 above)
Comment (ARoxdale) [reply meka/srcs/sound/sound.cpp]: It's mostly present from the old code. I wasn't sure at the time if this was affecting the sound output or not.

**Pull request closed without merging**